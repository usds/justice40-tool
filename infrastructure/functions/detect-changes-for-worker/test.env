REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1