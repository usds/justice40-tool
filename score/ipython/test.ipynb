{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
