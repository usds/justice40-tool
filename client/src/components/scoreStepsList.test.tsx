import * as React from 'react';
import {render} from '@testing-library/react';
import ScoreStepsList from './scoreStepsList';
import {LocalizedComponent} from '../test/testHelpers';

// TODO: Move this to a location that will detect on all tests
// See ticket: #550
beforeAll(() => {
  jest.spyOn(global.console, 'error').mockImplementation((...params) => {
    console.error(params);
  });
});

describe('rendering of the component', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <ScoreStepsList/>
      </LocalizedComponent>,
  );

  it('should match the snapshot of the MapIntroduction component', () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it('No console errors', () => {
    expect(console.error).toBeCalledTimes(0);
  });
});
