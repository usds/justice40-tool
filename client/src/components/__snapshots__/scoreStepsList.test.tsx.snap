// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the component should match the snapshot of the MapIntroduction component 1`] = `
<DocumentFragment>
  <h2>
    Methodology
  </h2>
  <div
    class="grid-row"
    data-testid="grid"
  >
    <div
      class="grid-col-7"
      data-testid="grid"
    >
      <p>
        The methodology for identifying communities of focus is calculated at the census block group level. Census block geographical boundaries are determined by the U.S. Census Bureau once every ten years. This tool utilizes the census block boundaries from 2010.
      </p>
      <p>
        The following describes the process for identifying communities of focus.
      </p>
    </div>
  </div>
  <ol
    class="usa-process-list"
  >
    <li
      class="usa-process-list__item"
    >
      <h3
        class="usa-process-list__heading"
        data-testid="processListHeading"
      >
        Gather datasets
      </h3>
      <p>
         
      </p>
      <p
        class="flush"
      >
        The methodology includes the following inputs that are equally weighted.
      </p>
      <h4>
        Percent of Area Median Income
      </h4>
      <p
        class="flush"
      >
        If a census block group is in a metropolitan area, this value is the median income of the census block group calculated as a percent of the metropolitan area’s median income.
      </p>
      <p>
        If a census block group is not in a metropolitan area, this value is the median income of the census block group calculated as a percent of the state’s median income.
      </p>
      <h4>
        Percent of households below or at 100% of the federal poverty line
      </h4>
      This is the percent of households in a state with a household income 
    below or at 100% of the 
      <a
        href="https://www.census.gov/topics/income-poverty/poverty/guidance/poverty-measures.html"
        rel="noreferrer"
        target="_blank"
      >
        federal poverty line
      </a>
      . This federal poverty line is calculated 
    based on the composition of each household (e.g., based on household size), but it does not vary geographically.
      <h4>
        The high school degree achievement rate for adults 25 years and older
      </h4>
      <p
        class="flush"
      >
        The percent of individuals who are 25 or older who have received a high school degree.
      </p>
    </li>
    <li
      class="usa-process-list__item"
    >
      <h3
        class="usa-process-list__heading"
        data-testid="processListHeading"
      >
        Determine communites of focus
      </h3>
      <p>
         
      </p>
      <p
        class="flush"
      >
        Under the existing formula, a census block group will be considered a community of focus if:
      </p>
      <p>
        (The median income is less than 80% of the area median income
      </p>
       
      <p
        class="flush"
      >
        OR
      </p>
       
      <p
        class="flush"
      >
        households living in poverty (at or below 100% of the federal poverty level) is greater than 20%)
      </p>
       
      <p
        class="flush"
      >
        AND
      </p>
       
      <p
        class="flush"
      >
        The high school degree achievement rate for adults 25 years and older is greater than 95%
      </p>
    </li>
  </ol>
</DocumentFragment>
`;
